libretro-common/queues/task_queue.c

```diff
@@ -91,6 +91,13 @@ static void task_queue_msg_push(retro_task_t *task,
 
 static void task_queue_push_progress(retro_task_t *task)
 {
+#ifdef HAVE_THREADS
+   /* msg_push callback interacts directly with the task properties (particularly title).
+    * make sure another thread doesn't modify them while rendering
+    */
+   slock_lock(property_lock);
+#endif
```

Comment:
That was my first assumption as well, but it mimics all the property functions:
https://github.com/libretro/RetroArch/blob/a96d12e8bcf4ea191a1fa4871f5c878ce1b9abac/libretro-common/queues/task_queue.c#L807-L816

If threads aren't enabled, it's just overkill. There wouldn't be any contention for the protected data.

Reply:
My concern is extra overhead on platforms with threaded support but not running threaded tasks.

A simple if statement to check that would be just a couple of instructions, but unnecessary lock primitive acquires/releases will be a lot more and can involve syscalls and/or thread context switches.

I guess it's fine if you are just following the example of the other functions, but those functions could certainly be optimized for non-threaded tasks.

Reply:
OK, so what would be the best way to address Cthulhu-throwaway's feedback? Can things be further optimized for non-threaded tasks?

Reply:
Those functions should have a regular and threaded implementation for the task queue's state machine implementation.
Someone should go through that code and refactor all functions that would differ between threaded and non-threaded implementations, implementing regular and threaded variations so it can work like a state machine.

Don't think it falls within this PR though and he is clearly following the same "convention" as the other functions.

Reply:
OK, so basically whatever you are talking about can be addressed in another PR? Would it be something you would personally like to look at, or is it something that should be left up to Jamiras in a followup PR?

So basically in a roundabout way I'm asking if all criticisms of this PR have been addressed, and if we can move to merging it.

Reply:
I think the PR is fine, but further unknown side effects should be watched for closely after merging it.

Optimization of task functions for non-threaded tasks doesn't fall to either of us. Someone with time and interest should look into it; I have my hands full and I don't feel like taking any more extra work.